### PR TITLE
Improvements on Impala query generation

### PR DESCRIPTION
**Pull request by espinielli**
This is still WIP.
It seems to work for the normal cases.

Also the query generated when serials and airport are specified seems ok: but veeeerrryyy slow when executed on my machine, so much that I gave up.

I need to understand how to intercept the logged request sent to Impala...I see the `logging.info` to spit it but I am too unexperienced to get it out ;-)

Strangely I have to force the commit because of a pre-commit hook check error in part of the code I did not touch:

```bash
black....................................................................Passed
flake8...................................................................Passed
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

traffic/data/eurocontrol/b2b/pkcs12.py:59:43: error: Argument 1 to
"from_cryptography_key" of "PKey" has incompatible type "Optional[Any]";
expected "Union[DSAPrivateKey, DSAPublicKey, RSAPrivateKey, RSAPublicKey]"
            crypto.PKey.from_cryptography_key(private_key)
                                              ^
Found 1 error in 1 file (checked 1 source file)

isort....................................................................Failed
- hook id: isort
- exit code: 1

Executable `isort` not found
```

### CODE REVIEW

File: traffic/data/adsb/opensky_impala.py [hunk 1]
```diff
@@ -755,15 +755,11 @@ def history(
...
-                "sv." if count_airports_params > 0 else "", icao24.lower()
-            )
+            other_params += "and sv.icao24='{}' ".format(icao24.lower())
 
         elif isinstance(icao24, Iterable):
             icao24 = ",".join("'{}'".format(c.lower()) for c in icao24)
-            other_params += "and {}icao24 in ({}) ".format(
-                "sv." if count_airports_params > 0 else "", icao24
-            )
+            other_params += "and sv.icao24 in ({}) ".format(icao24.lower())
```
Comment (espinielli) [inline traffic/data/adsb/opensky_impala.py:762]: This is needed anyway to be aligned with the one above

File: traffic/data/adsb/opensky_impala.py [hunk 2]
```diff
@@ -875,7 +866,7 @@ def history(
 
         cumul = []
         sequence = list(split_times(start, stop, date_delta))
-        columns = ", ".join(self._impala_columns)
```
Comment (espinielli) [inline traffic/data/adsb/opensky_impala.py:878]: This seemed completely redundant given that later in the original code it was completelt overwritten.

**Comment by espinielli** [comment]
Current code works, at least before 
```python
            df = self._format_dataframe(df)
            df = self._format_history(df, nautical_units=nautical_units)
```

Here some queries I tested against and that return a Pandas data frame:

```python
from traffic.data import opensky
import logging

logging.basicConfig(level=logging.INFO)

opensky.history(
    start="2021-08-24 00:00",
    stop="2021-08-24 01:00",
    airport="ESSA",
    limit=3,
    cached=False,
)

opensky.history(
    start="2021-08-24 00:00",
    stop="2021-08-24 01:00",
    arrival_airport="ESSA",
    limit=3,
    cached=False,
)

opensky.history(
    start="2021-08-24 00:00",
    departure_airport="ESSA",
    arrival_airport="EGLL",
    limit=3,
    cached=False,
)

opensky.history(
    start="2021-08-24 00:00",
    stop="2021-08-24 01:00",
    airport="ESSA",
    arrival_airport="EGLL",
    limit=3,
    cached=False,
)

opensky.history(
    start="2021-08-24 00:00",
    stop="2021-08-24 01:00",
    arrival_airport="ESSA",
    serials=-1408232560,
    limit=3,
    cached=False,
)

opensky.history(
    start="2021-08-24 00:00",
    stop="2021-08-24 01:00",
    serials=(-1408232560, -1408232534),
    limit=3,
    cached=False,
)

opensky.history(
    start="2021-08-24 00:00",
    stop="2021-08-24 01:00",
    departure_airport="ESSA",
    serials=(-1408232560, -1408232534),
    limit=3,
    cached=False,
)

```

File: traffic/data/adsb/opensky_impala.py [hunk 3]
```diff
@@ -825,6 +825,10 @@ def history(
             )
 
         elif arrival_airport is not None:
+            if airport is not None:
```
Comment (espinielli) [inline traffic/data/adsb/opensky_impala.py:828]: I understand that `arrival_airport`/`departure_airport` and `airport` cannot coexist, so I added the relevant error

File: traffic/data/adsb/opensky_impala.py [hunk 4]
```diff
@@ -825,6 +825,10 @@ def history(
...
+                raise RuntimeError(
+                    "airport may not be set if " "arrival_airport is set"
+                )
```
Comment (espinielli) [inline traffic/data/adsb/opensky_impala.py:831]: I understand that `arrival_airport`/`departure_airport` and `airport` cannot coexist, so I added the relevant error

File: traffic/data/adsb/opensky_impala.py [hunk 5]
```diff
@@ -838,6 +842,10 @@ def history(
             )
 
         elif departure_airport is not None:
+            if airport is not None:
+                raise RuntimeError(
+                    "airport may not be set if " "departure_airport is set"
+                )
```
Comment (espinielli) [inline traffic/data/adsb/opensky_impala.py:848]: I understand that `arrival_airport`/`departure_airport` and `airport` cannot coexist, so I added the relevant error

**Comment by espinielli** [comment]
I have now tested all the above and they work (i.e. do not throw errors ;-) even with `_format_dataframe` and `format_history`.

Still the tests are not passing but I guess this is due to having to resave the caches for the relevant test cases: but I am no expert here. So Xavier, feed me back.

File: traffic/data/adsb/opensky_impala.py (hunk 2 above)
Comment (xoolive) [reply traffic/data/adsb/opensky_impala.py:878]: It's only overwritten if `count_airports_params > 0` right?

Comment (espinielli) [reply traffic/data/adsb/opensky_impala.py:878]: The
```python
", ".join(f"sv.{field}" for field in self._impala_columns)
```
was repeated (overwriting whatever was done before, without the `sv.` prefix) if `count_airports_params > 0`.
But in the new code the  `sv.` prefix is always there and so I initialized `columns` with the prefix
and added the `est.` columns for the `count_airports_params > 0` case.

**Comment by xoolive** [comment]
Thanks a lot for all this work.

- It seems there is still some bug in `test_history()` but you should be able to run it now (try locally but it should also work on Github Actions).
- Based on the examples you mentioned in the PR, I add a `test_complex_queries()` test suite in `test_impala.py` but I think they deserve some real testing on the results. Would you mind editing them?

**Comment by xoolive** [comment]
Issues on hold (to share with @espinielli):
- [ ] pandas issue:
  `FAILED tests/test_impala.py::test_history - AttributeError: 'DatetimeTZBlock' object has no attribute 'view'`
- [ ] credentials issue from secrets
- [ ] queries with aliases seem to make Impala run out of memory

**Comment by espinielli** [comment]
The use of aliases is problematic.
The following test case (in `test_impala.py`) fails
```python
opensky.history(
            "2019-11-11 10:00",
            "2019-11-11 10:10",
            bounds=lfbo_tma,
            serials=1433801924,
            cached=False,
        )
```
But it generates a nice / correct query:
```sql
select
   sv.time, sv.icao24, sv.lat, sv.lon, sv.velocity, sv.heading, sv.vertrate,
   sv.callsign, sv.onground, sv.alert, sv.spi, sv.squawk, sv.baroaltitude,
   sv.geoaltitude, sv.lastposupdate, sv.lastcontact, sv.hour
from
   state_vectors_data4 as sv,
   state_vectors_data4.serials s
where
   sv.hour>=1573466400.0 and sv.hour<1573470000.0
   and sv.time>=1573466400.0 and sv.time<1573467000.0
   and s.ITEM = 1433801924
   and sv.lon>=0.11666666666666667 and sv.lon<=2.816666666666667
   and sv.lat>=42.361666666666665 and sv.lat<=44.537778333333335;
```

BUT it fails due to exceeding the memory scan allowed.
Running `EXPLAIN` on the query we get

```
[hadoop-29:21000] > explain select sv.time, sv.icao24, sv.lat, sv.lon, sv.velocity, sv.heading, sv.vertrate, sv.callsign, sv.onground, sv.alert, sv.spi, sv.squawk, sv.baroaltitude, sv.geoaltitude, sv.lastposupdate, sv.lastcontact, sv.hour from state_vectors_data4 as sv , state_vectors_data4.serials s  where sv.hour>=1573466400.0 and sv.hour<1573470000.0 and sv.time>=1573466400.0 and sv.time<1573467000.0 and s.ITEM = 1433801924 and sv.lon>=0.11666666666666667 and sv.lon<=2.816666666666667 and sv.lat>=42.361666666666665 and sv.lat<=44.537778333333335;
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Explain String                                                                                                                                                                         |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Max Per-Host Resource Reservation: Memory=0B                                                                                                                                           |
| Per-Host Resource Estimates: Memory=3.46GB                                                                                                                                             |
| WARNING: The following tables are missing relevant table and/or column statistics.                                                                                                     |
| opensky.state_vectors_data4                                                                                                                                                            |
|                                                                                                                                                                                        |
| PLAN-ROOT SINK                                                                                                                                                                         |
| |                                                                                                                                                                                      |
| 04:EXCHANGE [UNPARTITIONED]                                                                                                                                                            |
| |                                                                                                                                                                                      |
| 02:NESTED LOOP JOIN [CROSS JOIN, BROADCAST]                                                                                                                                            |
| |                                                                                                                                                                                      |
| |--03:EXCHANGE [BROADCAST]                                                                                                                                                             |
| |  |                                                                                                                                                                                   |
| |  01:SCAN HDFS [opensky.state_vectors_data4.serials s]                                                                                                                                |
| |     partitions=49140/49140 files=50052 size=14.37TB                                                                                                                                  |
| |     predicates: s.ITEM = 1433801924                                                                                                                                                  |
| |                                                                                                                                                                                      |
| 00:SCAN HDFS [opensky.state_vectors_data4 sv]                                                                                                                                          |
|    partitions=1/49140 files=1 size=397.50MB                                                                                                                                            |
|    predicates: sv.lat <= 44.537778333333335, sv.lat >= 42.361666666666665, sv.lon <= 2.816666666666667, sv.lon >= 0.11666666666666667, sv.time < 1573467000.0, sv.time >= 1573466400.0 |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

where you can see that there is a *HUGE* scan:

```
| |  01:SCAN HDFS [opensky.state_vectors_data4.serials s]                                                                                                                                |
| |     partitions=49140/49140 files=50052 size=14.37TB                                                                                                                                  |
| |     predicates: s.ITEM = 1433801924                                                                                                                                                  ```
and that is why the execution of the query bails out and fails the test.
If you execute the query in the Impala shell you get:

```
[hadoop-29:21000] > select sv.time, sv.icao24, sv.lat, sv.lon, sv.velocity, sv.heading, sv.vertrate, sv.callsign, sv.onground, sv.alert, sv.spi, sv.squawk, sv.baroaltitude, sv.geoaltitude, sv.lastposupdate, sv.lastcontact, sv.hour from state_vectors_data4 as sv , state_vectors_data4.serials s  where sv.hour>=1573466400.0 and sv.hour<1573470000.0 and sv.time>=1573466400.0 and sv.time<1573467000.0 and s.ITEM = 1433801924 and sv.lon>=0.11666666666666667 and sv.lon<=2.816666666666667 and sv.lat>=42.361666666666665 and sv.lat<=44.537778333333335;
WARNINGS: Memory limit exceeded: Error occurred on backend hadoop-220.lambda.armasuisse.ch:22000 by fragment c44695eef9cce360:3f9474cf00000008
Memory left in process limit: -263.34 MB
Query(c44695eef9cce360:3f9474cf00000000): Reservation=0 ReservationLimit=12.80 GB OtherMemory=11.23 GB Total=11.23 GB Peak=11.23 GB
  Fragment c44695eef9cce360:3f9474cf00000019: Reservation=0 OtherMemory=11.22 GB Total=11.22 GB Peak=11.22 GB
    NESTED_LOOP_JOIN_NODE (id=2): Total=11.21 GB Peak=11.21 GB
      Nested Loop Join Builder: Total=11.21 GB Peak=11.21 GB
    HDFS_SCAN_NODE (id=0): Total=8.00 KB Peak=8.00 KB
      Exprs: Total=4.00 KB Peak=4.00 KB
    EXCHANGE_NODE (id=3): Reservation=7.12 MB OtherMemory=0 Total=7.12 MB Peak=7.12 MB
      KrpcDeferredRpcs: Total=0 Peak=4.13 KB
    KrpcDataStreamSender (dst_id=4): Total=1.83 KB Peak=1.83 KB
    CodeGen: Total=1.59 KB Peak=511.50 KB
  Fragment c44695eef9cce360:3f9474cf00000008: Reservation=0 OtherMemory=10.66 MB Total=10.66 MB Peak=229.61 MB
    HDFS_SCAN_NODE (id=1): Total=10.51 MB Peak=229.48 MB
      Exprs: Total=4.00 KB Peak=4.00 KB
    KrpcDataStreamSender (dst_id=3): Total=102.38 KB Peak=102.38 KB
    CodeGen: Total=506.00 B Peak=159.50 KBProcess: memory limit exceeded. Limit=16.00 GB Total=16.26 GB Peak=16.26 GB
  Buffer Pool: Free Buffers: Total=398.82 MB
  Buffer Pool: Clean Pages: Total=0
  Buffer Pool: Unused Reservation: Total=-21.96 MB
  Data Stream Service Queue: Limit=819.20 MB Total=0 Peak=2.19 MB
  Data Stream Manager Early RPCs: Total=0 Peak=100.80 KB
  TCMalloc Overhead: Total=227.23 MB
  RequestPool=root.openskyguest: Total=3.41 GB Peak=14.86 GB
    Query(e14753aa05d36767:e7653f6900000000): Reservation=0 ReservationLimit=12.80 GB OtherMemory=178.17 MB Total=178.17 MB Peak=186.94 MB
    Query(145ded8363e054c:2e0066b400000000): Reservation=0 ReservationLimit=12.80 GB OtherMemory=2.91 GB Total=2.91 GB Peak=2.91 GB
    Query(4542db3ddde171b9:d40697a200000000): Reservation=0 ReservationLimit=12.80 GB OtherMemory=23.25 MB Total=23.25 MB Peak=28.71 MB
    Query(594371411a3b4b11:113c11400000000): Reservation=0 ReservationLimit=12.80 GB OtherMemory=88.27 MB Total=88.27 MB Peak=91.37 MB
    Query(39445857acdeeea8:f8f06f6100000000): Reservation=220.00 MB ReservationLimit=12.80 GB OtherMemory=1.14 MB Total=221.14 MB Peak=253.50 MB
  RequestPool=root.opensky: Total=0 Peak=592.47 MB
  RequestPool=root.axs: Total=0 Peak=106.05 MB
  RequestPool=root.openskypoweruser: Total=11.23 GB Peak=13.44 GB
    Query(c44695eef9cce360:3f9474cf00000000): Reservation=0 ReservationLimit=12.80 GB OtherMemory=11.23 GB Total=11.23 GB Peak=11.23 GB
  Untracked Memory: Total=1.03 GB

WARNING: The following tables are missing relevant table and/or column statistics.
opensky.state_vectors_data4
Memory limit exceeded: Error occurred on backend hadoop-220.lambda.armasuisse.ch:22000 by fragment c44695eef9cce360:3f9474cf00000008
Memory left in process limit: -263.34 MB
Query(c44695eef9cce360:3f9474cf00000000): Reservation=0 ReservationLimit=12.80 GB OtherMemory=11.23 GB Total=11.23 GB Peak=11.23 GB
  Fragment c44695eef9cce360:3f9474cf00000019: Reservation=0 OtherMemory=11.22 GB Total=11.22 GB Peak=11.22 GB
    NESTED_LOOP_JOIN_NODE (id=2): Total=11.21 GB Peak=11.21 GB
      Nested Loop Join Builder: Total=11.21 GB Peak=11.21 GB
    HDFS_SCAN_NODE (id=0): Total=8.00 KB Peak=8.00 KB
      Exprs: Total=4.00 KB Peak=4.00 KB
    EXCHANGE_NODE (id=3): Reservation=7.12 MB OtherMemory=0 Total=7.12 MB Peak=7.12 MB
      KrpcDeferredRpcs: Total=0 Peak=4.13 KB
    KrpcDataStreamSender (dst_id=4): Total=1.83 KB Peak=1.83 KB
    CodeGen: Total=1.59 KB Peak=511.50 KB
  Fragment c44695eef9cce360:3f9474cf00000008: Reservation=0 OtherMemory=10.66 MB Total=10.66 MB Peak=229.61 MB
    HDFS_SCAN_NODE (id=1): Total=10.51 MB Peak=229.48 MB
      Exprs: Total=4.00 KB Peak=4.00 KB
    KrpcDataStreamSender (dst_id=3): Total=102.38 KB Peak=102.38 KB
    CodeGen: Total=506.00 B Peak=159.50 KBProcess: memory limit exceeded. Limit=16.00 GB Total=16.26 GB Peak=16.26 GB
  Buffer Pool: Free Buffers: Total=398.82 MB
  Buffer Pool: Clean Pages: Total=0
  Buffer Pool: Unused Reservation: Total=-21.96 MB
  Data Stream Service Queue: Limit=819.20 MB Total=0 Peak=2.19 MB
  Data Stream Manager Early RPCs: Total=0 Peak=100.80 KB
  TCMalloc Overhead: Total=227.23 MB
  RequestPool=root.openskyguest: Total=3.41 GB Peak=14.86 GB
    Query(e14753aa05d36767:e7653f6900000000): Reservation=0 ReservationLimit=12.80 GB OtherMemory=178.17 MB Total=178.17 MB Peak=186.94 MB
    Query(145ded8363e054c:2e0066b400000000): Reservation=0 ReservationLimit=12.80 GB OtherMemory=2.91 GB Total=2.91 GB Peak=2.91 GB
    Query(4542db3ddde171b9:d40697a200000000): Reservation=0 ReservationLimit=12.80 GB OtherMemory=23.25 MB Total=23.25 MB Peak=28.71 MB
    Query(594371411a3b4b11:113c11400000000): Reservation=0 ReservationLimit=12.80 GB OtherMemory=88.27 MB Total=88.27 MB Peak=91.37 MB
    Query(39445857acdeeea8:f8f06f6100000000): Reservation=220.00 MB ReservationLimit=12.80 GB OtherMemory=1.14 MB Total=221.14 MB Peak=253.50 MB
  RequestPool=root.opensky: Total=0 Peak=592.47 MB
  RequestPool=root.axs: Total=0 Peak=106.05 MB
  RequestPool=root.openskypoweruser: Total=11.23 GB Peak=13.44 GB
    Query(c44695eef9cce360:3f9474cf00000000): Reservation=0 ReservationLimit=12.80 GB OtherMemory=11.23 GB Total=11.23 GB Peak=11.23 GB
  Untracked Memory: Total=1.03 GB (1 of 3 similar)
```

Now if you remove the `sv` alias you get the query:

```
select 
   time, icao24, lat, lon, velocity, heading, vertrate, 
   callsign, onground, alert, spi, squawk, baroaltitude, 
   geoaltitude, lastposupdate, lastcontact, hour 
from 
   state_vectors_data4, 
   state_vectors_data4.serials s
where
   hour>=1573466400.0 and hour<1573470000.0 
   and time>=1573466400.0 and time<1573467000.0 
   and s.ITEM = 1433801924 
   and lon>=0.11666666666666667 and lon<=2.816666666666667 
   and lat>=42.361666666666665 and lat<=44.537778333333335;
```
and the `EXPLAIN` on it says it all ok!
This time you scan 1 file instead of 49140!!!!
```
[hadoop-29:21000] > explain select time, icao24, lat, lon, velocity, heading, vertrate, callsign, onground, alert, spi, squawk, baroaltitude, geoaltitude, lastposupdate, lastcontact, hour from state_vectors_data4, state_vectors_data4.serials s  where hour>=1573466400.0 and hour<1573470000.0 and time>=1573466400.0 and time<1573467000.0 and s.ITEM = 1433801924 and lon>=0.11666666666666667 and lon<=2.816666666666667 and lat>=42.361666666666665 and lat<=44.537778333333335;
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Explain String                                                                                                                                                                                                    |
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Max Per-Host Resource Reservation: Memory=0B                                                                                                                                                                      |
| Per-Host Resource Estimates: Memory=1.46GB                                                                                                                                                                        |
| WARNING: The following tables are missing relevant table and/or column statistics.                                                                                                                                |
| opensky.state_vectors_data4                                                                                                                                                                                       |
|                                                                                                                                                                                                                   |
| PLAN-ROOT SINK                                                                                                                                                                                                    |
| |                                                                                                                                                                                                                 |
| 05:EXCHANGE [UNPARTITIONED]                                                                                                                                                                                       |
| |                                                                                                                                                                                                                 |
| 01:SUBPLAN                                                                                                                                                                                                        |
| |                                                                                                                                                                                                                 |
| |--04:NESTED LOOP JOIN [CROSS JOIN]                                                                                                                                                                               |
| |  |                                                                                                                                                                                                              |
| |  |--02:SINGULAR ROW SRC                                                                                                                                                                                         |
| |  |                                                                                                                                                                                                              |
| |  03:UNNEST [state_vectors_data4.serials s]                                                                                                                                                                      |
| |                                                                                                                                                                                                                 |
| 00:SCAN HDFS [opensky.state_vectors_data4]                                                                                                                                                                        |
|    partitions=1/49140 files=1 size=397.50MB                                                                                                                                                                       |
|    predicates: lat <= 44.537778333333335, lat >= 42.361666666666665, lon <= 2.816666666666667, lon >= 0.11666666666666667, time < 1573467000.0, time >= 1573466400.0, !empty(opensky.state_vectors_data4.serials) |
|    predicates on s: s.ITEM = 1433801924                                                                                                                                                                           |
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```
and the query runs till the end (I added limit just to reduce the output):

```
[hadoop-29:21000] > select time, icao24, lat, lon, velocity, heading, vertrate, callsign, onground, alert, spi, squawk, baroaltitude, geoaltitude, lastposupdate, lastcontact, hour from state_vectors_data4, state_vectors_data4.serials s  where hour>=1573466400.0 and hour<1573470000.0 and time>=1573466400.0 and time<1573467000.0 and s.ITEM = 1433801924 and lon>=0.11666666666666667 and lon<=2.816666666666667 and lat>=42.361666666666665 and lat<=44.537778333333335 limit 3;
+------------+--------+-------------------+-------------------+-------------------+-------------------+----------+----------+----------+-------+-------+--------+--------------+-------------+----------------+----------------+------------+
| time       | icao24 | lat               | lon               | velocity          | heading           | vertrate | callsign | onground | alert | spi   | squawk | baroaltitude | geoaltitude | lastposupdate  | lastcontact    | hour       |
+------------+--------+-------------------+-------------------+-------------------+-------------------+----------+----------+----------+-------+-------+--------+--------------+-------------+----------------+----------------+------------+
| 1573466749 | 345643 | 44.53715515136719 | 1.041673932756696 | 280.3795313698676 | 191.0006743428709 | 0        | VLG72W   | false    | false | false | 7606   | 10660.38     | 10728.96    | 1573466748.431 | 1573466748.997 | 1573466400 |
| 1573466750 | 345643 | 44.533447265625   | 1.040693010602679 | 280.3795313698676 | 191.0006743428709 | 0        | VLG72W   | false    | false | false | 7606   | 10668        | 10728.96    | 1573466749.978 | 1573466749.992 | 1573466400 |
| 1573466751 | 345643 | 44.53221130371094 | 1.040366036551339 | 280.8845395834225 | 190.9806500101735 | -0.32512 | VLG72W   | false    | false | false | 7606   | 10660.38     | 10721.34    | 1573466750.42  | 1573466750.958 | 1573466400 |
+------------+--------+-------------------+-------------------+-------------------+-------------------+----------+----------+----------+-------+-------+--------+--------------+-------------+----------------+----------------+------------+
```

Nasty!

**Comment by xoolive** [comment]
rel #154

**Comment by xoolive** [comment]
If credentials don't work from your github secrets, we may have to double check them together in your settings.

**Comment by espinielli** [comment]
At this stage, with the cache initially cleared, all Impala tests pass (and pass much faster once the cache is filled):

```
 espin   master → origin U:2  ❲c❳ traffic  ~/repos/traffic  poetry run pytest --log-level=DEBUG tests/test_impala.py
========================================================================================== test session starts ==========================================================================================
platform darwin -- Python 3.9.4, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /Users/espin/repos/traffic, configfile: pyproject.toml
plugins: cov-2.12.1, anyio-3.3.0
collected 4 items

tests/test_impala.py ....                                                                                                                                                                         [100%]

===================================================================================== 4 passed in 281.63s (0:04:41) =====================================================================================
 espin   master → origin U:2  ❲c❳ traffic  ~/repos/traffic  poetry run pytest --log-level=DEBUG tests/test_impala.py
========================================================================================== test session starts ==========================================================================================
platform darwin -- Python 3.9.4, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /Users/espin/repos/traffic, configfile: pyproject.toml
plugins: cov-2.12.1, anyio-3.3.0
collected 4 items

tests/test_impala.py ....                                                                                                                                                                         [100%]

========================================================================================== 4 passed in 43.81s ===========================================================================================
```

**Comment by espinielli** [comment]
Just temporarily disabled coverage test and triggered a new CI.

**Comment by xoolive** [comment]
Many thanks @espinielli 
What a trip!! :bicyclist: :tada:

**Pull request merged**